dask_jobqueue/remote_slurm.py

```diff
@@ -0,0 +1,174 @@
+import logging
+from contextlib import contextmanager
+from typing import Dict, Generator
+from urllib.parse import urljoin
+
+import aiohttp
+
+from .core import cluster_parameters, job_parameters
+from .slurm import SLURMCluster, SLURMJob, slurm_format_bytes_ceil
+
+logger = logging.getLogger(__name__)
+
+
+class RemoteSLURMJob(SLURMJob):
+    def __init__(self, *args, **kwargs):
+        self.api_client_session = kwargs.pop("api_client_session")
+        self.api_url = kwargs.pop("api_url")
+        self.remote_job_extra = (
+            kwargs.pop("remote_job_extra") if "remote_job_extra" in kwargs else {}
+        )
+        super().__init__(*args, **kwargs)
+
+    @contextmanager
+    def job_file(self):
+        """
+        Override the job_file and pass back the job_script. job_file() is used by Job.start and
+         gives output directly to SLURMJob._submit_job().
+        """
+        yield self.job_script()
+
+    @property
+    def _job_configuration(self) -> Dict:
```

Comment:
Seeing as this function is the exact same form as in SlurmJob it's probably worth it to create a dataclass which can then be transformed to ways (as dict and as sbatch script).

@jacobtomlinson what do you think?